### PR TITLE
ARTEMIS-1443 Fix issue starting Artemis in OSGI with CORE protocol

### PR DESCRIPTION
**Pull request by ryeats**
Fixed several issues with starting Artemis in OSGI you can now not specify CORE as a protocol in the artemis.xml and have it successfully start. Also allowed Artemis to start when no protocols are specified although only the CORE protocol will be available if no other acceptor has any protocols defined.

### CODE REVIEW

File: artemis-features/src/main/resources/artemis.xml [hunk 1]
```diff
@@ -22,28 +22,68 @@ under the License.
...
+         xsi:schemaLocation="urn:activemq:core ">
+
+      <persistence-enabled>true</persistence-enabled>
+
+      <!-- this could be ASYNCIO, MAPPED, NIO
+           ASYNCIO: Linux Libaio
+           MAPPED: mmap files
+           NIO: Plain Java Files
+       -->
+      <journal-type>NIO</journal-type>
```
Comment (clebertsuconic) [inline artemis-features/src/main/resources/artemis.xml]: Why NIO?

**Comment by clebertsuconic** [comment]
How to reproduce the issue?

**Comment by clebertsuconic** [comment]
I believe this is fixing something.. but I need to how reproduce it exactly.. maybe add a test case...

I will hold merging this until you provide me some details.

**Comment by ryeats** [comment]
If you add ?protocols=CORE to the artemis.xml rebuild and then run ArtemisFeatureTest it will fail because artemis doesn't start. Since I added that in this PR it will be tested going forward.

Also Ill explain more since their could be a better solution. CORE protocol isn't registered as a service factory like the other protocols its just [manually registered](https://github.com/apache/activemq-artemis/blob/master/artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/server/impl/RemotingServiceImpl.java#L150-L152). This causes an issue in the ProtocolTracker since OSGi will see the other Protocols that are registered using the java ServiceLoader but since CORE isn't ProtocolTracker will never start artemis because it is waiting for CORE.

**Pull request closed without merging**